nemo/core/classes/modelPT.py

```diff
@@ -178,7 +202,12 @@ def setup_optimization(self, optim_config: Optional[Union[DictConfig, Dict]] = N
 
             logging.info("Optimizer config = %s", str(optimizer))
 
-            return optimizer
+            self._optimizer = optimizer
+            # return optimizer
+
+        self._scheduler = prepare_lr_scheduler(
```

Comment:
There are a few more return optimizers above inside the function, so please also add this lines there.

Reply:
fixed. thanks for catching that!